### DISCUSSION TITLE
Unbridged Camera

### DISCUSSION THREAD

**Discussion by clossie** [dgreif/ring#1478] [General]
I have noted the following statement in the current version release notes referring to the option to Unbridge Cameras: **if true, all Ring Cameras we be treated as external accessories, which generally leads to better performance. This means they each need to be individually added to HomeKit** 

I thought I would try selecting the option to see what happens - In alignment with the notes, my 4 cameras dropped from the Homebrodge plugin and are no longer visible in HomeKit...**If your cameras are already bridged, they will be deleted from HomeKit when you enable this option, and you will need to reconfigure any associated automations or HomeKit settings**

What I cannot seem to figure out is how to add the cameras back in? Any guidance would be appreciated as I would rather figure this out before it becomes default

**Comment by gcastelli3** [comment]
i was told by the programmer.  "Simply select to add an accessory, and then, on the screen where there is the option to scan the code, select "more options" and you will see the individual cameras. Select the one you wish to add and enter the code or select use camera to use the QR code."  I can see the devices in that list however, I have tried both the qr code (camera) and the 5 digit code and neither has worked for 
 for me.  Hoping this will be fixed soon.  I am going back to original setup for now.

**Reply by tsightler** [reply]
It is unlikely there is anything to fix, the unbridged code has been there for quite some time and is known to work.  If you are unable to add the devices it is likely something related to your specific setup.  There is another report here https://github.com/dgreif/ring/issues/1461 with a similar sounding issue, but the user hasn't come back.

I have a suspicion that it could be related to running on Windows, perhaps something with Windows firewall. as accessories bind to their own ports.

Alternatively, it could be related to mDNS, what mDNS advertiser are you using?

**Reply by gcastelli3** [reply]
Not sure what you mean by advertiser the router is an Xfinity router and my Hoobs is a Hoobs Pro.   I also checked the router security warnings and it is not telling me that there is an issue there.   I am willing to try any suggestions you have.

**Reply by tsightler** [reply]
Ah, you listed HB version, not HOOBS version, in your issue so I did not originally notice it was HOOBS.  You should contact HOOBS support for any issues as, overall, HOOBS is somewhat outdated and a bit of a black box.  Perhaps they set firewalls on non-standard ports or something and maybe they don't allow selecting the [mDNS advertiser](https://github.com/homebridge/homebridge/wiki/mDNS-Options).

**Reply by gcastelli3** [reply]
Thanks.  I will try that.  But just so I understand is it the mDNS advertiser that would not recognize a QR code in Home?  When I go through the process you described to me it never recognizes a QR code and the 5 digit pin for Ring is xxxxx and when entering it in Home manually it shows up as xxxx-x.  Not sure if any of this matters?

**Reply by tsightler** [reply]
It's not really clear what you mean by never recognizing the code.  I mean, the code is just a minor security tool used for pairing, you never mentioned the actual error that is generated.  I would expect the code would be 8 digits, not 5 anyway.   Regardless, you should contact HOOBS support for any issues with HOOBS.

**Reply by gcastelli3** [reply]
I will try but I doubt that they will be of any help as the cameras appear in Hoobs just not able to add them to Home.  All Ring pin codes are just 5 digits.  Will I be able to continue using your plugin as long as I set bridgecameras to false even after future udates?

**Reply by tsightler** [reply]
I'm not sure why codes are only 5 digits, that is weird but I guess maybe a HOOBS thing.  This is why it's not really easy to provide direct support to HOOBS users as they basically repackage Homebridge and do various things to make it easier to consume, but with that they take on responsibility to be first line support

In the end, this is not a HOOBS plugins, it is a Homebridge plugin.  The goal is to support HOOBS as well, but they have to take point support because only they are familiar with the changes they make and the configuration of their solution, that information is a black box to us.   Some plugin developers have even explicitly stopped supporting HOOBS because of this as HOOBS users come and flood their Github pages with issues that are not related to the plugin at all, vs contacting HOOBS support.

At some point, support for bridged cameras will likely be removed, but there is no current timeline.

**Reply by gcastelli3** [reply]
Trying to dig into this. Maybe the problem is that I am trying to enter the Ring pin but what may be needed is the bridge pin which is 8 digits. It does allow me to add Accessory Port Pool range a well. In the logs it tells me what ports are assigned for each camera as well

**Reply by tsightler** [reply]
To be completely honest, I have no idea what you are referring to with regards to Ring PIN.  In base homebridge you get a QR code/pin for the bridge so that you can add it.  This code is 8 digits in the format of XXX-XX-XXX.  You just use this same PIN for pairing standalone accessories as well.  When the plugin starts up you can look in the logs and should see messages similar to this:

```
Front Door 717B is running on port 42316.
Please add [Front Door 717B] manually in Home app. Setup Code: 649-41-545
```

The setup code is the PIN, and should be the same for every camera as well as the bridge, at least it is in native Homebridge.  I would not really expect HOOBS to be different.

**Reply by gcastelli3** [reply]
OK that was it.  Home was looking for the plugin pin and not the pin of the Ring device.  They are now appearing and connected correctly.  The Hoobs log does not provide the Setup code line as shown but it is on the Ring Bridge page under PIN.  Thanks, sorry for driving you crazy but hopefully this will help others as well.

**Reply by tsightler** [reply]
Great!  Glad you got it working and thanks for sharing!

**Comment by clossie** [comment]
Following the above - I solved my issue as follows. 

System: 
Homebridge plugin running on QNAP T251
Ring Plugin - Running as Child Bridge

Process - 
Select Unbridge Camera in Ring Plugin. 
Restart the Ring Child Bridge.
Restart the Homebridge server. 
Open the Ring Plugin logs and ensure each camera you have shows as a Manually Configure option. The code (for me anyway) was the same as the overarching Homebridge 8 digit pin.
Restart your HomeKit Homehub - for me a MiniPod
In HomeKit select “Add Accessory” and select the “more options” text.
My 4 cameras now appeared herein. Selected each and used the Homebridge plugin code and one by one they added fine.

**Reply by Bward02** [reply]
For anyone having this problem I was able to solve mine by changing my mDNS settings back to Avahi. This allowed my homekit to find the cameras. .
